src/memory.rs

```diff
@@ -111,7 +108,23 @@ impl MemoryInstance {
     ///
     /// [`LINEAR_MEMORY_PAGE_SIZE`]: constant.LINEAR_MEMORY_PAGE_SIZE.html
     pub fn alloc(initial: Pages, maximum: Option<Pages>) -> Result<MemoryRef, Error> {
-        validate_memory(initial, maximum).map_err(Error::Memory)?;
+        {
+            use std::convert::TryInto;
+            let initial_u32: u32 = initial
+                .0
+                .try_into()
+                .map_err(|_| Error::Memory("initial can't be coerced to u32".into()))?;
```

Comment:
Value of what "can't be coerced" would be great

Reply:
Makes sense!